src/stores/activityStore.ts

```diff
@@ -1,9 +1,29 @@
-import {observable} from "mobx";
+import {action, observable, makeAutoObservable} from "mobx";
 import { createContext } from "react";
+import {io} from "socket.io-client";
+import {apiUrl} from "../config/api";
+import {CardArrayInterface, changeCardNaming} from "../Components/utils/changeCardNaming";
+const socket = io(`${apiUrl}`);
 
-class ActivityStore{
-    @observable title: string[] = [];
-    @observable loadingInitial = false;
-}
+export class ActivityStore{
+    constructor() {
+        makeAutoObservable(this, {}, {autoBind: true});
+    }
+    @observable commonCards = ['', '', '', '', ''];
+    @observable playersNewCards: string[][] = [['','']];
+
+    @action updateCards = async () => {
+        socket.emit('turn', {}, (response: CardArrayInterface[]) => {
+            const newCommonCards = changeCardNaming([response]);
+            this.commonCards = newCommonCards[0] as string[];
 
-export default createContext(new ActivityStore())
+        });
```

Comment:
Hmm I get why you did this but let's keep the store simpler and move the socket emitting to another place.
Concern of this Store should be to set correct values. Fetching/Emitting data is another concern.

If you would keep the `updateCards` as a simple action, this could be later used for other reasons
```typescript
@action
async setCommonCards(newCards: SomeCardInterface) {
    this.commonCards = newCards;
}
```

Now what about socket emitting? First of all the function by itself could be extracted to a separate function (remember single responsibility?)
```typescript
async function emitAnotherTurn(): Promise<yourCardInterface> {
   return new Promise(res => {
     socket.emit('turn', {}, (response: CardArrayInterface[]) => {
            const newCommonCards = changeCardNaming([response]);
            res(newCommonCards[0] as string[]);
        });
   })
}
```
now instead of using updateCards you can just emit and then save value

```typescript
const updateCards = async (e: React.MouseEvent<HTMLButtonElement> & { target: { value: string; }; }) => {
        e.preventDefault();
        const newCommonCards = await emitAnotherTurn();
        gameFlow.setCommonCards(newCommonCards)
    }

```